code/game/objects/structures/signs/sign_eyechart.dm

```diff
@@ -28,11 +28,8 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/sign/eyechart, 32)
 		. += span_warning("...But it's too dark to make out anything.")
 		return
 
-	var/colorblind = FALSE
-	if(iscarbon(user))
-		var/mob/living/carbon/carbon_user = user
-		colorblind = !!carbon_user.has_trauma_type(/datum/brain_trauma/mild/color_blindness)
-	else if(isdog(user) || iscat(user))
+	var/colorblind = HAS_TRAIT(user, TRAIT_COLORBLIND)
+	if(isdog(user) || iscat(user))
 		colorblind = TRUE // i know these pets are not colorblind in the conventional sense, but it's an easter egg, ok?
```

Comment:
If we're adding a trait we might as well apply it to the mobs as innate traits no?